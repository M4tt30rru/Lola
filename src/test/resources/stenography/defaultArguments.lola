##Import "std.lola"
##Find(DefaultableArgumentDeclaration)
 ##Either ##ArgumentDeclaration(_dec) ##or ##ArgumentDeclaration(_dec) = ##Expression(_value)
##Find ##Type (_type) ##Identifier(_name) (##NoneOrMore ##DefaultableArgumentDeclaration(_dec) ##separator ,) ##Block(_block)
 ##replace ##ForEach([i for i, d in enumerate(_decs) if hasattr(d, '_value')] + [len(_decs)]) ##(_type) ##(_name) (##(','.join(d._dec.name for d in _decs[:_]))) ##(_block.name if _ == len(_decs) else '{return ' + _name.name + '(' + ','.join(d._dec._name.name for d in _decs[:_]) + ',' + ','.join(d._value.name for d in _decs[_:]) + ');}')

int f(int a, int b, int c=1, int d=a+b) {
 return a + b * c - d;
}
