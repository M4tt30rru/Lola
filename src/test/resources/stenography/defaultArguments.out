int f(int a, int b) {
 return f(a, b, 1, a+b);
}
int f(int a, int b, int c) {
 return f(a, b, c, a+b);
}
int f(int a, int b, int c, int d) {
 return a + b * c - d;
}
